src/get-started/java-binding.md

```diff
@@ -880,56 +873,320 @@ Currently Java Binding includes the following built-in services:
+will be used for communication between nodes.
+
+```$sh
+$ exonum-java generate-config \
+    testnet/common.toml \
+    testnet \
+    --no-password \
+    --peer-address 127.0.0.1:5400
 ```
 
-where possible values for `service-name` are:
+### Step 3. Finalize Configuration
 
-- `configuration` for Configuration Update Service
-- `btc-anchoring` for Anchoring Service
-- `time` for Time Oracle
+On third, and the last step we combine private part of the node configuration
+and the public parts of each node of the network.
 
-!!! note
-    In case there is no such section,
-    only Configuration Service will be activated.
+We provide a path to a private part of the node configuration, a path to a
+resulting node configuration file and a list of paths to public configuration
+files of every node.
 
-Below is the sample of the `services.toml` file that enables
-all possible built-in Exonum services and two user services:
+```$sh
+$ exonum-java finalize \
+    testnet/sec.toml \
+    testnet/node.toml \
+    --public-configs testnet/pub.toml
+```
 
-```toml
-system_services = ["configuration", "btc-anchoring", "time"]
-[user_services]
-service_name1 = "/path/to/service1_artifact.jar"
-service_name2 = "/path/to/service2_artifact.jar"
+After completing each of the steps, the `testnet/node.toml` file contains a
+final node configuration and can be used to run the node with specified
+parameters.
+
+### Running the Node
+
+Unlike configuration process, `run` command of Exonum Java App is different from
+similar Exonum Rust command.
+
+There are several required parameters here:
+
+- `--db-path` for a path to the database directory.
+- `--node-config` for a path to the final node configuration file.
+- `--artifacts-path` for a path to the directory containing compiled service
+  artifacts.
+- `--ejb-port` for port that Java services will use for communication.
+  Java Binding does not use public API address directly.
+- `--public-api-address` and `--private-api-address` for socket addresses for
+  node user API. Public API address is used by users to send transactions and
+  requests to the blockchain, private one is used by node administrators to
+  perform different maintenance actions. Setting these addresses isn't strictly
+  necessary for every node, but we will need available API ports for deploying
+  and starting service using `exonum-launcher`.
+
+There are also optional parameters useful for debugging purposes, logging
+configuration and JVM fine tuning:
+
+- `--jvm-args-prepend` and `--jvm-args-append`: Additional parameters for JVM
+  that prepend and append the rest of arguments. Must not have a leading dash.
+  For example, `Xmx2G`.
+- `--jvm-debug`: Allows JVM being remotely debugged over the JDWP protocol.
+  Takes a socket address as a parameter in form of `HOSTNAME:PORT`. For example,
+  `localhost:8000`.
+- `--ejb-log-config-path` for path to Log4j 2 configuration file. Default config
```

Comment:
I think we shall link an appropriate section on logging here; and logging to this section.